Memories/Memories/VCs/AddViewController.swift

```diff
@@ -0,0 +1,159 @@
+    var coordinates: CLLocationCoordinate2D? { didSet { finishSaving() }}
+    private let context = CIContext(options: nil)
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        picker.delegate = self
+        UserDefaults.standard.removeObject(forKey: "CurrentAudio")
+        UserDefaults.standard.removeObject(forKey: "CurrentVideo")
+    }
+    
+    @IBAction func textFieldReturn(_ sender: UITextField) {
+        sender.resignFirstResponder()
+    }
+    
+    @IBAction func addImage(_ sender: Any) {
+        picker.allowsEditing = true
+        let actionSheet = UIAlertController(title: "Photo Source", message: "Choose a source", preferredStyle: .actionSheet)
+        
+        actionSheet.addAction(UIAlertAction(title: "Camera", style: .default , handler: { (sction: UIAlertAction) in
+            if UIImagePickerController.isSourceTypeAvailable(.camera) {
+                self.picker.sourceType = .camera
+                self.present(self.picker, animated: true, completion: nil) }
+            else { print("Camera not available") } }))
+        actionSheet.addAction(UIAlertAction(title: "Photo Libary", style: .default , handler: { (sction: UIAlertAction) in
+            self.picker.sourceType = .photoLibrary
+            self.present(self.picker, animated: true, completion: nil) }))
+        actionSheet.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+        self.present(actionSheet, animated: true)
+    }
+    
+    @IBAction func filterImage(_ sender: Any) {
+        filterButton.isSelected.toggle()
+        if filterButton.isSelected {
+        guard let image = originalImage else { return }
+        var ciImage = CIImage(data: image)
+        let filter = CIFilter.colorMonochrome()
+        filter.inputImage = ciImage
+        filter.color = CIColor(color: .black)
+        ciImage = filter.outputImage
+        guard let outputCI = ciImage else { return }
+        guard let outputCG = context.createCGImage(outputCI, from: CGRect(origin: CGPoint.zero, size: imageView.image!.size)) else { return }
+        let uiImage = UIImage(cgImage: outputCG)
+        imageView.image = uiImage
+            self.image = uiImage.pngData()
+        }
+        else {
+            imageView.image = UIImage(data: originalImage!)
+        }
+    }
+    
+    @IBAction func done(_ sender: Any) {
+        save()
+    }
+    
+    private func save() {
+        //Check for title
+        guard let title = titleField.text, !(titleField.text?.isEmpty ?? true) else { Helper.alert(on: self, "Please add a title", "You can't save this without one"); return }
+        titleText = title
+        //Get location
+        locationManager.delegate = self
+        locationManager.requestWhenInUseAuthorization()
+        guard CLLocationManager.locationServicesEnabled() else {
+            Helper.alert(on: self, "Please turn on access for your location", "We are unable to display this data to you otherwise"); return }
```

Comment:
Don't hide a long line's closing `}`

If it's short like `return nil` I'm ok with it being on one line, but not when it's this huge expression. Split up long lines so they are readable.